### ISSUE TITLE
Please check that you are not altering your HTTP REFERER.

### ISSUE THREAD

**Issue by caywin25** [FreshRSS/FreshRSS#2989]
Bonsoir à tous,

Merci pour votre travail.
J'ai installé freshrss,
Je dispose d'un proxy qui renvoie vers un serveur web sur lequel j'ai installé freshrss.
Tout va bien mais pendant l'installation toutes les étapes passent sauf la dernière : 
> Please check that you are not altering your HTTP REFERER.

Que veut dire cette étape et comment régler le soucis svp ?
Merci pour votre aide.

**Comment by Alkarex** [comment]
Bonsoir,
Essayez avec un autre navigateur sans extension. Peut-être avez-vous un réglage/extension de votre navigateur qui modifie le champ HTTP *Referer*.
Sinon, cela peut aussi venir d'un proxy mal réglé qui n'envoie pas tous les entêtes HTTP. Personnellement, j'aime bien [Traefik](https://docs.traefik.io/) comme proxy (français !).

**Comment by caywin25** [comment]
J'ai testé de tester avec deux autres navigateurs + désactivations de toutes les extensions, même soucis :(

**Comment by Alkarex** [comment]
Quel est le proxy ? Quelle plateforme ?
Pouvez-vous tester sans passer par votre proxy ?

**Comment by caywin25** [comment]
Le proxy est nginx sous debian.
C'est difficile sans celui-ci.

**Comment by caywin25** [comment]
Ah je pense avoir trouvé

**Comment by caywin25** [comment]
Je clos cette issue, le soucis cest réglé. Merci

**Comment by Alkarex** [comment]
Quel était le problème ?

**Comment by nomad-57** [comment]
I've had the same erorr while installing via docker(linuxserverio image).
The problem only seems to occur if the installation in done while freshrss is run under a proxy.
I successfully installed it without the proxy and then proxying afterwards did not give any errors.

**Comment by aledeg** [comment]
@caywin25 It could be nice to add the full description of the problem and how to solve it in the documentation! At least if you could explain it here, it would be appreciated. Thank you

**Comment by hesch** [comment]
Okay, so I encountered this problem today. My setup is as follows:
nginx-container that handles TLS and acts as a reverse proxy.
freshrss-container from the freshrss/freshrss:latest image.

From my digging I found the check in this function: https://github.com/FreshRSS/FreshRSS/blob/a626e5fff8d1cfcc7280396bd8b17b177b6345d2/lib/lib_rss.php#L396-L411

In my setup, the variables get set like this:
```php
$_SERVER['HTTP_HOST'] == <container_name>
$_SERVER['SERVER_NAME'] == <container_name>
$_SERVER['HTTPS'] == ''
```
Without any further configuration in the nginx reverse proxy, it just transparently forwards the HTTP_REFERER (so it is set to https://<hostname>/). Maybe this function should be extended to handle X-Forwarded-For headers? (as mentioned here: https://github.com/FreshRSS/FreshRSS/issues/565#issuecomment-61605204)

This check is only done to warn the user while installation. So here is a workaround to get around this check for Installation in Firefox (I just temporarily disabled the Referer-Header):
1. Enter `about:config`
2. Set `network.http.sendRefererHeader` to 0
3. Complete the install
4. Reset `network.http.sendRefererHeader`

I am not sure if I am now susceptible to XSRF attacks, but this comment seems to suggest, that the Referer check is no longer needed? https://github.com/FreshRSS/FreshRSS/issues/554#issuecomment-239632084
Maybe @Alkarex can clear up, if the check is even needed or what the purpose is exactly.
